mzLib/MassSpectrometry/SpectraFileInfo.cs

```diff
@@ -3,91 +3,214 @@
+        {
+            if (obj is null) return false;
+            if (obj is SpectraFileInfo otherSpecInfo)
+            {
+                return Equals(otherSpecInfo);
+            }
+            if (obj is ISampleInfo otherSampleInfo)
+            {
+                return Equals(otherSampleInfo);
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns a hash code based on all identity components: <see cref="FullFilePathWithExtension"/>,
+        /// <see cref="Condition"/>, <see cref="BiologicalReplicate"/>, <see cref="Fraction"/>, 
+        /// and <see cref="TechnicalReplicate"/>.
+        /// </summary>
+        /// <returns>A hash code for this instance.</returns>
         public override int GetHashCode()
         {
             return HashCode.Combine(
                 StringComparer.Ordinal.GetHashCode(FullFilePathWithExtension ?? string.Empty),
                 StringComparer.Ordinal.GetHashCode(Condition ?? string.Empty),
                 BiologicalReplicate,
-                TechnicalReplicate,
-                Fraction);
+                Fraction,
+                TechnicalReplicate);
         }
 
+        /// <summary>
+        /// Returns the file name (with extension) extracted from <see cref="FullFilePathWithExtension"/>.
+        /// </summary>
+        /// <returns>The file name portion of the full file path.</returns>
         public override string ToString()
         {
             return Path.GetFileName(FullFilePathWithExtension);
         }
+
+        /// <summary>
+        /// Compares this instance to another <see cref="ISampleInfo"/> for sorting purposes.
+        /// Comparison order: <see cref="FullFilePathWithExtension"/>, <see cref="Condition"/>,
+        /// <see cref="BiologicalReplicate"/>, <see cref="Fraction"/>, <see cref="TechnicalReplicate"/>.
+        /// All comparisons use ascending order (A before B for strings, 1 before 2 for numbers).
+        /// Non-null values sort before null.
+        /// </summary>
+        /// <param name="other">The other <see cref="ISampleInfo"/> to compare to.</param>
+        /// <returns>
+        /// A negative value if this instance precedes <paramref name="other"/>;
+        /// zero if they are equal; a positive value if this instance follows <paramref name="other"/>.
+        /// </returns>
+        public int CompareTo(ISampleInfo? other)
+        {
+            // Non-null comes before null
+            if (other is null) return -1;
+
+            if (other is SpectraFileInfo) 
+            {
+                // Check for equality returns 0 if equal.
+                if (Equals(other)) return 0;
+
+                // Compare by FullFilePathWithExtension (A before B)
+                int comparison = string.Compare(FullFilePathWithExtension ?? string.Empty, other.FullFilePathWithExtension ?? string.Empty, StringComparison.Ordinal);
+                if (comparison != 0) return comparison;
+
```

Comment:
The comparison by full file path should come last. For lfq, the file paths will never be equivalent, so the rest of the code is unused.